functions.php

```diff
@@ -12,14 +9,121 @@
 function getWeaponsData(object $db) : array
 {
     $query = $db->prepare("SELECT `name`,`damage_per_second`, 
-       `rate_of_fire`, `ammo_comsumption`, `ammo_efficiency`, 
+       `rate_of_fire`, `ammo_consumption`, `ammo_efficiency`, 
        `default_weapon_number`, `image_url` FROM `weapons`;");
     $query->execute();
 
     return $query->fetchAll();
 }
 
+/**
+ * Returns true if the weapon has a name, returns false otherwise
+ *
+ * @param array $weapon the weapon to check if it has a name
+ * @return bool returns true if the weapon has a name
+ */
+function validateWeaponName(array $weapon) : bool {
+    return array_key_exists('name',$weapon);
+}
+
+/**
+ * Validates the weapon data and returns true if the weapon data has the expected format
+ *
+ * @param array $weaponData the weapon data to validate
+ * @return bool returns true if the weapon data has the expected format
+ */
+function validateWeaponData(array $weaponData) : bool {
+
+    foreach ($weaponData as $weaponDatum) {
+
+        if (!(gettype($weaponDatum['name']) == 'string' &&
+            gettype($weaponDatum['damage_per_second']) == 'string' &&
+            gettype($weaponDatum['rate_of_fire']) == 'string' &&
+            gettype($weaponDatum['ammo_consumption']) == 'string' &&
+            gettype($weaponDatum['ammo_efficiency']) == 'string' &&
+            gettype($weaponDatum['default_weapon_number']) == 'string' &&
+            gettype($weaponDatum['image_url']) == 'string')
+        )
+        {
+            return false;
+        }
+    }
+    return true;
+}
+
+/**
+ * Returns the weapon name
+ *
+ * @param array $weapon the weapon whose name will get returned
+ * @return string the name of the weapon
+ */
+function displayWeaponName(array $weapon) : string
+{
+    return '<li>' . $weapon['name'] . '</li>';
+}
+
+/**
+ * Returns a bullet-pointed list of the names of each piece of weapon data.
+ *
+ * @param array $data the data that will get returned.  This data must have a 'name' key.
+ * @return string returns the bullet-pointed list of data names.
+ */
+function displayWeaponNames(array $data) : string
+{
+    $result = '';
+    foreach($data as $datum)
+    {
+            if(validateWeaponName($datum))
```

Comment:
Probably don't need to create entire function for this. Just use array_key_exists as above in the function declaration & add a comment to say what you are doing & then you are not calling your own functions within functions.